### DISCUSSION TITLE
Launch multiple matchmaking challenges simultaneously

### DISCUSSION THREAD

**Discussion by ppipelin** [lichess-bot-devs/lichess-bot#883] [Ideas]
Hello,
Would it be possible to add an option which allows the program to launch multiple matchmaking simultaneously? e.g. `matchmaking>concurrency`. It would be interesting mostly for engines that are not multithreaded.
Best,

**Comment by MarkZH** [comment]
Lichess puts a limit on how many challenges per day a bot can request. Allowing multiple challenges at a time would cause bots to hit the limit faster.

Why do you want to launch multiple simultaneous challenges?

**Reply by ppipelin** [reply]
This would be interesting for bots that plays longer time control like for 1h games, but I understand your point.

**Reply by MarkZH** [reply]
That makes sense. We'll consider it, but we have to do some planning since the matchmaking routine assumes only one challenge is in play at a time.

**Comment by MarkZH** [comment]
I've added a PR that you can test out: #913. Basically, if a game lasts longer than 10 minutes, then another matchmaking challenge will be created. This requires setting `matchmaking: allow_during_games: true` in your configuration file.

**Reply by ppipelin** [reply]
Thanks, I'll test it out. Here https://lichess.org/@/RadianceEngine for 1800+1 & 1800+3

**Reply by MarkZH** [reply]
That PR has been merged, so I'm closing this discussion. Feel free to reopen or file an issue if there are problems.
